include/xtensor-sparse/xmap_container.hpp

```diff
@@ -0,0 +1,278 @@
+#ifndef XTENSOR_SPARSE_LIL_HPP
+#define XTENSOR_SPARSE_LIL_HPP
+
+#include <array>
+#include <map>
+#include <utility>
+#include <vector>
+
+#include <xtensor/xstorage.hpp>
+#include <xtensor/xexception.hpp>
+#include <xtensor/xtensor.hpp>
+#include <xtensor/xadapt.hpp>
+#include <xtensor/xio.hpp>
+
+namespace xt
+{
+    template<class T>
+    class xmap_container
+    {
+    public:
+
+        using value_type = T;
+        using const_reference = const T&;
+        using reference = T&;
+        using pointer = T*;
+        using const_pointer = const T*;
+        using size_type = std::size_t;
+        using difference_type = std::ptrdiff_t;
+
+        using shape_type = svector<size_type>;
+        using strides_type = svector<size_type>;
+
+        using inner_shape_type = svector<size_type>;
+
+        xmap_container(const shape_type& shape);
+
+        size_type size() const noexcept;
+
+        const shape_type& shape() const;
+        size_type dimension() const;
+        const strides_type& strides() const;
```

Comment:
I still think we should not expose the strides. They are just a convenience for computing new keys when reshaping.